src/callback_handler/mod.rs

```diff
@@ -71,14 +78,15 @@ impl CallbackHandlerBuilder {
         if self.shutdown_channel.is_none() {
             return Err(anyhow!("shutdown_channel_rx not provided"));
         }
+        if self.fulcio_and_rekor_data.is_none() {
+            return Err(anyhow!("fulcio_and_rekor_data not provided"));
+        }
```

Comment:
This line, together with the one you added below don't match my **personal** style. Feel free to ignore this comment, I would however do something like that:

```rust
let fulcio_and_rekor_data = self.fulcio_and_rekor_data.ok_or_else(|| anyhow!("fulcio_and_rekor_data not provided"))?;

// later on
let sigstore_client = sigstore_verification::Client::new(
            self.oci_sources.clone(),
            self.oci_sources.clone(),
            self.docker_config.clone(),
            self.docker_config.clone(),
            &fulcio_and_rekor_data,
            fulcio_and_rekor_data,
        )?;
```

Reply:
I like that approach! it looks more idiomatic, and I don't like to see unwrap(). I changed also shutdown_channel, so it is done in the same way https://github.com/kubewarden/policy-evaluator/pull/117/commits/7327c5ab91a0338a0922a719a8e200698be754db